src/widgets/Dialogs.js

```diff
@@ -285,6 +287,8 @@ define(function (require, exports, module) {
      * @return {Dialog}
      */
     function showModalDialogUsingTemplate(template, autoDismiss) {
+        BrambleEvents.triggerDialogOpened();
```

Comment:
I think you should move this down inside the `one("shown", function () {...` handler.